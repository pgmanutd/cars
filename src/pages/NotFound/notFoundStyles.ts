import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      paddingTop: theme.spacing(7),
      paddingBottom: theme.spacing(7),
      textAlign: 'center',
    },
    appLogoContainer: {
      marginBottom: theme.spacing(2),
    },
  }),
);
