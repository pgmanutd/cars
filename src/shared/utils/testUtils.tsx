import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
