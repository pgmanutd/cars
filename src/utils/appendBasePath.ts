const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
