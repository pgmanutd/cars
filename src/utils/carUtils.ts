import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
