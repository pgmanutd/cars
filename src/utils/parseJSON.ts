const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
