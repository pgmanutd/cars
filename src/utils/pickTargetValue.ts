import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
