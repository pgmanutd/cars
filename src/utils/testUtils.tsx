import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  ...render(
    <HelmetProvider>
      <Router history={history}>
        <Route
          path={path}
          render={routeProps => React.cloneElement(ui, { ...routeProps })}
        />
      </Router>
    </HelmetProvider>,
  ),
  history,
});
