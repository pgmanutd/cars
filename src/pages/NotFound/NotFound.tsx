import React from 'react';
import { Helmet } from 'react-helmet-async';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Box from '@material-ui/core/Box';
import Link from '@material-ui/core/Link';
import { Link as RouterLink } from 'react-router-dom';

import routePaths from 'constants/routePaths';

import { useTranslate } from 'features/Translate';
import AppLogo from 'features/AppLogo';

import { useStyles } from './notFoundStyles';

const NotFound: React.FC = () => {
  const classes = useStyles();
  const { translate } = useTranslate();

  return (
    <Container
      data-testid="NotFound"
      component="section"
      className={classes.root}
    >
      <Helmet>
        <title>{translate('pages.NotFound.documentTitle')}</title>
      </Helmet>
      <Box className={classes.appLogoContainer}>
        <AppLogo />
      </Box>
      <Typography variant="h4" gutterBottom>
        {translate('pages.NotFound.headerText')}
      </Typography>
      <Typography variant="body1" gutterBottom>
        {translate('pages.NotFound.bodyText')}
      </Typography>
      <Typography variant="body1" gutterBottom>
        {translate('pages.NotFound.redirect.text')}{' '}
        <Link component={RouterLink} to={routePaths.cars} color="textSecondary">
          {translate('pages.NotFound.redirect.link')}
        </Link>
        .
      </Typography>
    </Container>
  );
};

export default NotFound;
