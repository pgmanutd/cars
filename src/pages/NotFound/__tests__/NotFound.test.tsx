import React from 'react';
import { wait } from '@testing-library/react';

import { renderWithProviders } from 'utils/testUtils';

import NotFound from '../NotFound';

describe('<NotFound />', () => {
  const setup = () => {
    const { renderResult } = renderWithProviders(<NotFound />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it('should render document title "404 - Not Found"', async () => {
    setup();

    await wait(() => expect(document.title).toEqual('404 - Not Found'));
  });
});
