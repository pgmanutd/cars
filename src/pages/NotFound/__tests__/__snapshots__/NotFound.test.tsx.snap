// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <section
    class="MuiBox-root MuiBox-root-2"
    data-testid="NotFound"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      404 - Not Found
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Sorry, the page you are looking for does not exist.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      You can always go back to the 
      <a
        href="/"
      >
        homepage
      </a>
      .
    </p>
  </section>
</DocumentFragment>
`;
